### PR TITLE
Closes #259: fqdn&siteUrl_to_settingservicebean

### LINKED ISSUE #259: Move fqdn and siteUrl from system.getProperty to settingsservicebean property

### PR DESCRIPTION
**Pull request by dbojanek**

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/FileDownloadServiceBean.java [hunk 1]
```diff
@@ -73,7 +68,7 @@
     @Inject
     WorldMapPermissionHelper worldMapPermissionHelper;
     @Inject
-    FileDownloadHelper fileDownloadHelper;
+    SettingsServiceBean settingsService;
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/FileDownloadServiceBean.java]: Should be private

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/HandlenetServiceBean.java [hunk 2]
```diff
@@ -254,7 +254,7 @@ public String getSiteUrl() {
         if (hostUrl != null && !"".equals(hostUrl)) {
             return hostUrl;
         }
-        String hostName = systemConfig.getFqdn();
+        String hostName = settingsService.getValueForKey(SettingsServiceBean.Key.FQDN);
         if (hostName == null) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/HandlenetServiceBean.java]: Can it ever be null if we have default value?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/UrlManagerServiceBean.java [hunk 3]
```diff
@@ -20,7 +21,7 @@
     private UrlManager urlManager = new UrlManager();
 
     @Inject
-    private SystemConfig systemConfig;
+    SettingsServiceBean settingsService;
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/UrlManagerServiceBean.java]: Should be private

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/UrlManagerServiceBean.java [hunk 4]
```diff
@@ -172,7 +173,7 @@ public String getHostnamePlusBaseUrlPath(String url) throws SwordError {
             optionalPort = ":" + port;
         }
         String requestedHostname = u.getHost();
-        String hostName = systemConfig.getFqdn();
+        String hostName = settingsService.getValueForKey(SettingsServiceBean.Key.FQDN);
         if (hostName == null) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/UrlManagerServiceBean.java]: As above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractSubmitToArchiveCommand.java [hunk 5]
```diff
@@ -43,6 +43,11 @@ public DatasetVersion execute(CommandContext ctxt) throws CommandException {
             }
         }
 
+        String setting = ctxt.settings().getValueForKey(SettingsServiceBean.Key.FQDN);
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractSubmitToArchiveCommand.java]: Why were those lines needed here? Previously there was nothing about fqdn and siteurl here.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java [hunk 6]
```diff
@@ -1206,12 +1211,12 @@ private static void saveJsonToDisk(String datasetVersionAsJson) throws IOExcepti
      * "official" hostname. If none of these options is set,
      * defaults to the InetAddress.getLocalHOst() and https;
      */
-    private static String getDataverseSiteUrl() {
-        String hostUrl = System.getProperty(SITE_URL);
+    private static String getDataverseSiteUrl(String Fqdn, String SiteUrl) {
+        String hostUrl = SystemConfig.getDataverseSiteUrlStatic(Fqdn, SiteUrl);
         if (hostUrl != null && !"".equals(hostUrl)) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java]: Same as above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java [hunk 7]
```diff
@@ -1206,12 +1211,12 @@ private static void saveJsonToDisk(String datasetVersionAsJson) throws IOExcepti
...
             return hostUrl;
         }
-        String hostName = System.getProperty(FQDN);
+        String hostName = Fqdn;
         if (hostName == null) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java]: Same as above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/Xrecord.java [hunk 8]
```diff
@@ -148,12 +147,12 @@ private boolean isExtendedDataverseMetadataMode(String formatName) {
         return DATAVERSE_EXTENDED_METADATA_FORMAT.equals(formatName);
     }
 
-    private String getDataverseSiteUrl() {
-        String hostUrl = System.getProperty(SITE_URL);
+    private String getDataverseSiteUrl(String Fqdn, String SiteUrl) {
+        String hostUrl = SiteUrl;
         if (hostUrl != null && !"".equals(hostUrl)) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/Xrecord.java]: Same as above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/Xrecord.java [hunk 9]
```diff
@@ -148,12 +147,12 @@ private boolean isExtendedDataverseMetadataMode(String formatName) {
...
             return hostUrl;
         }
-        String hostName = System.getProperty(FQDN);
+        String hostName = Fqdn;
         if (hostName == null) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/Xrecord.java]: Same as above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/sitemap/SiteMapServiceBean.java [hunk 10]
```diff
@@ -2,17 +2,24 @@
...
 import javax.ejb.Asynchronous;
 import javax.ejb.Stateless;
+import javax.inject.Inject;
 import java.util.List;
 
 @Stateless
 public class SiteMapServiceBean {
 
+    @Inject
+    SettingsServiceBean settingsService;
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/sitemap/SiteMapServiceBean.java]: Should be private

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java [hunk 11]
```diff
@@ -198,11 +182,11 @@ public int getMinutesUntilPasswordResetTokenExpires() {
      * by the Settings Service configuration.
      */
     public String getDataverseSiteUrl() {
-        String hostUrl = System.getProperty(SITE_URL);
+        String hostUrl = settingsService.getValueForKey(SettingsServiceBean.Key.SiteUrl);
         if (hostUrl != null && !"".equals(hostUrl)) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: Same as above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java [hunk 12]
```diff
@@ -198,11 +182,11 @@ public int getMinutesUntilPasswordResetTokenExpires() {
...
             return hostUrl;
         }
-        String hostName = getFqdn();
+        String hostName = settingsService.getValueForKey(SettingsServiceBean.Key.FQDN);
         if (hostName == null) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: Same as above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/passwordreset/PasswordResetServiceBean.java [hunk 13]
```diff
@@ -47,6 +48,9 @@
     @Inject
     private SystemConfig systemConfig;
 
+    @Inject
+    SettingsServiceBean settingsService;
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/passwordreset/PasswordResetServiceBean.java]: Should be private

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/passwordreset/PasswordResetServiceBean.java [hunk 14]
```diff
@@ -105,7 +109,7 @@ public PasswordResetInitResponse requestPasswordReset(BuiltinUser aUser, boolean
     private String createResetUrl(PasswordResetData passwordResetData) {
         // default to localhost
         String finalHostname = "localhost";
-        String configuredHostname = systemConfig.getFqdn();
+        String configuredHostname = settingsService.getValueForKey(SettingsServiceBean.Key.FQDN);
         if (configuredHostname != null) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/passwordreset/PasswordResetServiceBean.java]: Same as above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DuraCloudSubmitToArchiveCommand.java [hunk 15]
```diff
@@ -121,7 +125,7 @@ public void run() {
                             new Thread(() -> {
                                 try (PipedOutputStream out = new PipedOutputStream(in)) {
                                     // Generate bag
-                                    BagGenerator bagger = new BagGenerator(new OREMap(dv, false, SystemConfig.getDataverseSiteUrlStatic(), LocalDate.now()), dataciteXml);
+                                    BagGenerator bagger = new BagGenerator(new OREMap(dv, false, SystemConfig.getDataverseSiteUrlStatic(fqdn, siteUrl), LocalDate.now()), dataciteXml);
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DuraCloudSubmitToArchiveCommand.java]: Should be changed to non-static

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolHandler.java [hunk 16]
```diff
@@ -78,14 +80,14 @@ public String getQueryParametersForUrl() {
...
-    private String getQueryParam(String key, String value) {
+    private String getQueryParam(String key, String value, String Fqdn, String SiteUrl) {
         ReservedWord reservedWord = ReservedWord.fromString(value);
         switch (reservedWord) {
             case FILE_ID:
                 // getDataFile is never null because of the constructor
                 return key + "=" + getDataFile().getId();
             case SITE_URL:
-                return key + "=" + SystemConfig.getDataverseSiteUrlStatic();
+                return key + "=" + SystemConfig.getDataverseSiteUrlStatic(Fqdn, SiteUrl);
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolHandler.java]: Should be changed to non-static

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/sitemap/SiteMapUtil.java [hunk 17]
```diff
@@ -92,7 +92,7 @@ public static void updateSiteMap(List<Dataverse> dataverses, List<Dataset> datas
 
             Element loc = document.createElement("loc");
             String dataverseAlias = dataverse.getAlias();
-            loc.appendChild(document.createTextNode(SystemConfig.getDataverseSiteUrlStatic() + "/dataverse/" + dataverseAlias));
+            loc.appendChild(document.createTextNode(SystemConfig.getDataverseSiteUrlStatic(Fqdn, SiteUrl) + "/dataverse/" + dataverseAlias));
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/sitemap/SiteMapUtil.java]: Should be changed to non-static

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/sitemap/SiteMapUtil.java [hunk 18]
```diff
@@ -116,7 +116,7 @@ public static void updateSiteMap(List<Dataverse> dataverses, List<Dataset> datas
 
             Element loc = document.createElement("loc");
             String datasetPid = dataset.getGlobalId().asString();
-            loc.appendChild(document.createTextNode(SystemConfig.getDataverseSiteUrlStatic() + "/dataset.xhtml?persistentId=" + datasetPid));
+            loc.appendChild(document.createTextNode(SystemConfig.getDataverseSiteUrlStatic(Fqdn, SiteUrl) + "/dataset.xhtml?persistentId=" + datasetPid));
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/sitemap/SiteMapUtil.java]: Should be non-static

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java [hunk 19]
```diff
@@ -1206,12 +1211,12 @@ private static void saveJsonToDisk(String datasetVersionAsJson) throws IOExcepti
      * "official" hostname. If none of these options is set,
      * defaults to the InetAddress.getLocalHOst() and https;
      */
-    private static String getDataverseSiteUrl() {
-        String hostUrl = System.getProperty(SITE_URL);
+    private static String getDataverseSiteUrl(String Fqdn, String SiteUrl) {
+        String hostUrl = SystemConfig.getDataverseSiteUrlStatic(Fqdn, SiteUrl);
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java]: Should be removed probably.

**Review by MrK191 — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/HandlenetServiceBean.java (hunk 2 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/HandlenetServiceBean.java]: This method is not used. Got rid of it.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractSubmitToArchiveCommand.java (hunk 5 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractSubmitToArchiveCommand.java]: Rather than passing siteUrl and new parameter I wanted to use `requestedSettings` We can discuss this approach.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java (hunk 6 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java]: Method removed.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java (hunk 7 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java]: Same as above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/Xrecord.java (hunk 8 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/Xrecord.java]: Same as above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/Xrecord.java (hunk 9 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/Xrecord.java]: Same as above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java (hunk 11 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: With `StringUtils.isNotBlank(hostUrl)` we check (just in case) if property is present.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java (hunk 12 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: Same as above

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/passwordreset/PasswordResetServiceBean.java (hunk 14 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/passwordreset/PasswordResetServiceBean.java]: I'll do the `isNotBlank` check here

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java (hunk 19 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java]: Method removed

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java (hunk 11 above)
Comment (MrK191) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: But if the value is blank it will return null and not empty string. I think we can assume that we will operate on default values and no-one with internal server access will try to sabotage the configuration on the server.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/passwordreset/PasswordResetServiceBean.java (hunk 14 above)
Comment (MrK191) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/passwordreset/PasswordResetServiceBean.java]: Above https://github.com/CeON/dataverse/pull/267#discussion_r308727877

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java (hunk 11 above)
Comment (MrK191) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: Ah nvm this method returns empty string.

Comment (MrK191) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: But I'm still thinking, if there is any reason to set this setting to blank?

Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: The bottom line is: if we don't then there is no need for this method whatsoever and we can just replace it with `settingsService.getValueForKey(SettingsServiceBean.Key.SiteUrl);`

Comment (MrK191) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: But this method doesn't only return SiteUrl so what do you mean?

Comment (MrK191) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: After discussion with Krzysiek we decided that getting rid of FQDN would be cleaner. You can use this method with implementation as you said `settingsService.getValueForKey(SettingsServiceBean.Key.SiteUrl);`. Or you can get rid of it and use settingsWrapper directly in xhtml.

Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: I've decided to keep `systemConfig.getDataverseSiteUrl()` for mostly aesthetic reasons. Code looks cleaner. 
IMO If we want to change all (many) calls to `systemConfig.getDataverseSiteUrl()` to `settingsService.getValueForKey(SettingsServiceBean.Key.SiteUrl)` we would introduce some unnecessary redundancy.

File: dataverse-webapp/src/main/resources/config/dataverse.default.properties [hunk 20]
```diff
@@ -84,6 +84,8 @@ TimerServer=false
 
 HideSchemaDotOrgDownloadUrls=false
 MinutesUntilPasswordResetTokenExpires=60
+SiteUrl=http://localhost:8080
+FQDN=localhost
```
Comment (MrK191) [inline dataverse-webapp/src/main/resources/config/dataverse.default.properties]: We should get rid of this key.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java [hunk 21]
```diff
@@ -233,38 +200,11 @@ public static String getFilesDirectoryStatic() {
...
-        String fqdn = System.getProperty(FQDN);
-        if (fqdn == null) {
-            try {
-                fqdn = InetAddress.getLocalHost().getCanonicalHostName();
-            } catch (UnknownHostException e) {
-                return null;
-            }
-        }
-        return fqdn;
+        return settingsService.getValueForKey(SettingsServiceBean.Key.FQDN);
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: We should get rid of this key.

**Review by MrK191 — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DuraCloudSubmitToArchiveCommand.java [hunk 22]
```diff
@@ -42,6 +42,7 @@
     private static final String DURACLOUD_PORT = ":DuraCloudPort";
     private static final String DURACLOUD_HOST = ":DuraCloudHost";
     private static final String DURACLOUD_CONTEXT = ":DuraCloudContext";
+    private static final String DATAVERSE_URL = "dataverseUrl";
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DuraCloudSubmitToArchiveCommand.java]: We already have the enum that can be used, there is no need to create redundant constant value.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractSubmitToArchiveCommand.java [hunk 23]
```diff
@@ -43,6 +43,9 @@ public DatasetVersion execute(CommandContext ctxt) throws CommandException {
             }
         }
 
+        requestedSettings.put("dataverseUrl", ctxt.systemConfig().getDataverseSiteUrl());
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractSubmitToArchiveCommand.java]: Same as below.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/UrlManagerServiceBean.java [hunk 24]
```diff
@@ -172,10 +176,8 @@ public String getHostnamePlusBaseUrlPath(String url) throws SwordError {
             optionalPort = ":" + port;
         }
         String requestedHostname = u.getHost();
-        String hostName = systemConfig.getFqdn();
-        if (hostName == null) {
-            hostName = "localhost";
-        }
+        String hostName = systemConfig.getDataverseServer();
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/UrlManagerServiceBean.java]: You can just use SiteUrl, getting the url, stripping it, then creating it again below is redundant.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java [hunk 25]
```diff
@@ -233,38 +202,16 @@ public static String getFilesDirectoryStatic() {
...
-            }
-        }
-        hostUrl = "https://" + hostName;
-        return hostUrl;
-    }
-
     /**
      * The "official" server's fully-qualified domain name:
      */
     public String getDataverseServer() {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java:200]: Since it's a our method now we should create test for it.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java [hunk 26]
```diff
@@ -198,24 +184,7 @@ public int getMinutesUntilPasswordResetTokenExpires() {
...
-                return null;
-            }
-        }
-        hostUrl = "https://" + hostName;
-        return hostUrl;
-    }
-
-    public String getFqdn() {
-        return System.getProperty(SystemConfig.FQDN);
+        return settingsService.getValueForKey(SettingsServiceBean.Key.SiteUrl);
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java:179]: Since it's a our method now we should create test for it.

**Review by MrK191 — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/UrlManagerServiceBean.java (hunk 24 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/UrlManagerServiceBean.java]: Not entirely true. `systemConfig.getDataverseServer()` returns only hostname without port, which is being taken from argument passed to the method.

Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/UrlManagerServiceBean.java]: We've agreed on simplifying this method to just return `systemConfig.getDataverseSiteUrl() + SwordConfigurationConstants.BASE_URL_PATH_CURRENT`

**Review by MrK191 — APPROVED** [review]
